src/s3ql/backends/gdrive.py

```diff
@@ -0,0 +1,345 @@
+'''
+gdrive.py - this file is part of S3QL.
+
+Copyright © 2016 Max Khon <fjoe@samodelkin.net>
+
+This work can be distributed under the terms of the GNU GPLv3.
+'''
+
+from ..logging import logging # Ensure use of custom logger class
+from ..inherit_docstrings import (copy_ancestor_docstring, ABCDocstMeta)
+from .common import (AbstractBackend, DanglingStorageURLError, NoSuchObject)
+import apiclient
+import base64
+import httplib2
+import io
+import json
+import oauth2client.client
```

Comment:
This is actually was the point where I decided to not reuse s3c implementation and use google python API: s3ql's oauth2 implementation uses OAuth 2.0 for devices flow. Google Drive API scope is not allowed for this flow: https://developers.google.com/identity/protocols/OAuth2ForDevices#allowedscopes